.github/workflows/link-branch-to-issue.yml

```
@@ -0,0 +1,51 @@
+name: Link Branch to Issue Workflow
+
+# This workflow triggers on creation of a new ref (branch or tag)
+on:
+  create:
+    branches: '*'  # Only triggers on branch creation
```

actionlint (1.7.4)

"branches" filter is not available for create event. it is only for merge_group, push, pull_request, pull_request_target, workflow_run events

(events)